crc/services/user_service.py

```diff
@@ -97,6 +101,12 @@ def stop_impersonating():
             session.delete(admin_session)
             session.commit()
 
+    @staticmethod
+    def get_impersonator():
+        if UserService.user_is_admin() and UserService.admin_is_impersonating():
```

Comment:
This seems brittle - why are you checking for user_is_admin() here?  Is it possible that user_is_admin would return false if we are impersonating a non-admin user?